src/response.js

```diff
@@ -8,61 +8,55 @@ export class DropboxResponse {
   }
 }
 
-export function parseResponse(res) {
+function throwIfError(res) {
   const clone = res.clone();
-
   if (!res.ok) {
-    return res.text()
+    return res.json()
       .then((data) => {
         // eslint-disable-next-line no-throw-literal
         throw {
           error: data,
           response: res,
           status: res.status,
         };
+      }, () => {
```

Comment:
It's better to do `then(() => {}).catch(() => {})` instead of `then(() => {}, () => {})`.

Reply:
Catch will actually not work in this case. We are looking to catch the error from `.json()`. If we add a `.catch`, it will catch the error we just threw which would just add a lot more convoluted logic into there. I did verify that the second parameter of then is for the promise rejection so it should work (although it should never even be hit unless we push bad error parsing on the server side)